### PR TITLE
Add ConnectorSession parameter to getInfo method

### PR DESCRIPTION
**Pull request by pajaks**
## Description
ConnectorSession is needed as some properties may be retrieved only from session information

## Release notes

( x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
Could you mention the motivation in PR description?

**Comment by martint** [comment]
I’d like to understand why this is needed. The design of the API is such that the table handles are obtained from Metadata, and the session information is provided at that time. The table handles should encapsulate all the information necessary to produce the result of the getInfo call.

**Comment by findepi** [comment]
I think pretty much every ConnectorMetadata call should be given the session
is there any particular reason not to provide it for getInfo?

File: core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java [hunk 1]
```diff
@@ -425,7 +425,7 @@ public Optional<Object> getInfo(Session session, TableHandle handle)
         CatalogHandle catalogHandle = handle.getCatalogHandle();
         ConnectorMetadata metadata = getMetadata(session, catalogHandle);
 
-        return metadata.getInfo(handle.getConnectorHandle());
+        return metadata.getInfo(session.toConnectorSession(), handle.getConnectorHandle());
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java:428]: `.toConnectorSession(catalog)`

File: core/trino-spi/pom.xml [hunk 2]
```diff
@@ -885,6 +885,12 @@
                                     <old>method void io.trino.spi.security.SystemSecurityContext::&lt;init&gt;(io.trino.spi.security.Identity, java.util.Optional&lt;io.trino.spi.QueryId&gt;)</old>
                                     <new>method void io.trino.spi.security.SystemSecurityContext::&lt;init&gt;(io.trino.spi.security.Identity, io.trino.spi.QueryId, java.time.Instant)</new>
                                 </item>
+                                <item>
+                                    <ignore>true</ignore>
+                                    <code>java.method.numberOfParametersChanged</code>
```
Comment (findepi) [inline core/trino-spi/pom.xml:890]: You can avoid that easily. Just leave old getInfo as deprecated and add new one delegating back to the old.

**Comment by martint** [comment]
Because there should be nothing that getInfo does that should vary based on the session, which should be the same session that was passed when obtaining the table handle in the first place.

File: core/trino-spi/pom.xml (hunk 2 above)
Comment (electrum) [reply core/trino-spi/pom.xml:890]: Yes, please leave the old one with `@Deprecated(forRemoval = true)`

**Pull request closed without merging**